nexus/src/app/mod.rs

```diff
@@ -81,6 +122,12 @@ pub struct Nexus {
     /// Task representing completion of recovered Sagas
     recovery_task: std::sync::Mutex<Option<db::RecoveryTask>>,
 
+    /// External dropshot servers
+    external_servers: Mutex<ExternalServers>,
+
+    /// Internal dropshot server
+    internal_server: Mutex<Option<DropshotServer>>,
```

Comment:
It seems like `external_servers` here is a `tokio::sync::Mutex` rather than `std::sync::Mutex` because `refresh_tls_config` sometimes holds the lock open across await points -- is that correct? If so could you document this in a comment?

Also does the same apply to `internal_server`? If not then maybe it makes sense to use a `std::sync::Mutex` instead (similar to `recovery_task` above?)

(I'm sure you're already familiar with it, but for folks who might be wondering about the difference between `std::sync::Mutex` and `tokio::sync::Mutex`: https://docs.rs/tokio/latest/tokio/sync/struct.Mutex.html#which-kind-of-mutex-should-you-use)

Reply:
Yeah, I did this for `external_server`, and only used the tokio mutex on the internal server for parity. It's true, however -- we don't need to use an async mutex here. Updated in e34fb02cba788ff3f9730134fcb6b936b52a87a2